keps/sig-storage/3333-reconcile-default-storage-class/README.md

```diff
@@ -0,0 +1,711 @@
+- Manually test version skew between the API server and KCM. See the expected behavior below in Version Skew Strategy.
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag.
+- Two versions passed since introducing the functionality that deprecates the
+  flag (to address version skew).
+- Address feedback on usage/changed behavior, provided on GitHub issues.
+- Deprecate the flag.
+
+### Upgrade / Downgrade Strategy
+
+No change in cluster upgrade / downgrade process.
+
+### Version Skew Strategy
+
+This feature is implemented only in the API server and KCM and controlled by
+`RetroactiveDefaultStorageClass` feature gate. Following cases may happen:
+
+| API server | KCM | behavior                                                                                                                         |
+|------------|-----|----------------------------------------------------------------------------------------------------------------------------------|
+| off | off | Existing Kubernetes behavior.                                                                                                    |
+| on | off| Existing Kubernetes behavior, only users can change `pvc.spec.storageClassName=nil` to a SC name.                                |
+| off | on | PV controller may try to change `pvc.spec.storageClassName=nil` to a new default SC name, which will fail on the API server. (*) |
+| on | on | New behavior.                                                                                                                    |
+
+*) For this case, we strongly suggest that the feature is enabled in the API
+server first and then in KCM. Similarly, the feature should be disabled in KCM
+first and then in the API server. This follows generic Kubernetes version skew
+support.
+
+## Production Readiness Review Questionnaire
+
+<!--
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
```

Comment:
Please fill this required PRR section for Alpha.

Reply:
Added.